File: leet/sherSquares.py

import math
def sherSquares(a,b):
	count = 0
	for k in range(a,b+1):
		original = math.sqrt(k)
		deformed = math.floor(original)
		if original == deformed:
			count=count + 1
	return count

t = int(raw_input())
arr = []
for i in range(t):
	dude = raw_input().split()
	dude = [int(x) for x in dude]
	ans = sherSquares(dude[0],dude[1])
	print ans

File: leet/sherSquares.cpp
#include <iostream>
#include <vector>
#include <math.h>

using namespace std;

int dude (int x, int y)
{

	int count = 0;
	for (int i = x; i<=y; i++)
	{
		double original = sqrt(i);
		double deformed = floor(original);
		
		if (original == deformed)
			count++;
	}
	return count;

}

int main()
{

	vector<int> v1;
	int t;
	cin >> t;
	for (int i = 0; i<t; i++)
	{
			int x;
			int y;
			cin >> x >> y;
			int ans;
			ans = dude(x,y);
			cout << ans << endl;
		


}
return 0;
}
